one acute rewording request, one milder one, and short elaboration on refactoring (#244)
CaptainSifff/paper_teaching-learning-RSE: institutionalised_education.md

# Required Next Steps
We have identified the RSE as an individual who brings a unique set of skills for

> supporting and undertaking modern digitally enhanced research into research groups and teams.

Reviewer: see discussion above

> supporting and undertaking modern digitally enhanced research into research groups and teams.

Reviewer 2: so in order to be consistent, I have to do this as 
```suggestion
supporting and undertaking modern software-enabled research into research groups and teams.
```

special knowledge in classes and projects they choose.
For example, statistics curricula can
be used to showcase RSE infrastructure, e.g. the R programming language and

> its collection of statistics libraries and integrated development environments

Reviewer 3: I think ecosystem is better term here, as collection of libraries implies a pile of unconnected tools. The point of R is that provides this interconnected ecosystem

Author: In a nutshell, the term "ecosystem" differs from "system" in that it focuses on systems having living organisms in it, and is a biology term. Software libraries and IDEs aren't. "Ecosystem" within these contexts is business/marketing speak which crept in the last decades and brings no new concept of any sort in the field. If you had studied computer science in an engineering department in the early 2000s like I did (with English-language books from those years), you would have found none of this use of the word "ecosystem". If there is the desire to adhere to a growing trend, I can understand that. But I maintain that is poor use of the language. 

I also do not claim to have suggested the best alternatives: for me it was to avoid the -- IMHO -- worst.

Moreover on your comment, @mhagdorn, in "The point of R is that provides this interconnected ecosystem": "interconnected" already is the property linking the "members" of the "ecosystem" within it -- "interconnected ecosystems" sounds like "bread made of dough", which is nonsense.

Author: I had documented myself on this wording issue before yesterday, so here from my notes.. 
To the best of my knowledge the term "ecosystem" appeared in 1935 in an article by Arthur George Tansley; the article I link here (https://doi.org/10.1177/0309133307083297, also from
 https://www.researchgate.net/profile/Stephen-Trudgill/publication/249823757_Tansley_AG_1935_The_use_and_abuse_of_vegetational_concepts_and_terms_Ecology_16_284_307/links/55a8e68708ae481aa7f6d8f9/Tansley-AG-1935-The-use-and-abuse-of-vegetational-concepts-and-terms-Ecology-16-284-307.pdf  ) treat that first appearance of the term.

Reviewer 4: > In a nutshell, the term "ecosystem" differs from "system" in that it focuses on systems having living organisms in it, and is a biology term. Software libraries and IDEs aren't.

I do believe that the term also applies in software and accurately describes the following:

- Software is a "living organism", since it grows, evolves, and "dies" (gets abandoned). Like a flower or a community of animals, it can grow uncontrollably and die sooner, or, with some help, it can grow in a more sustainable way.
- Decisions in one software project affect decisions in other software projects. Projects are not only interconnected in terms of functionality, but also in terms of mutual success, prosperity, and sustainability.

I love metaphors and I have no issue with borrowing or misusing terms from other domains and contexts, if it makes communication more effective.

I agree with @mhagdorn that in this case, this term is just the most accurate.

Reviewer 2: Remember, that this part still has to be very much written...
So for today I'd stick with the ecosystem, and basically postpone this until this one is closer to maturity.

> its collection of statistics libraries and integrated development environments

Reviewer 2: ```suggestion
its ecosystem of statistics libraries and integrated development environments
```

the Common European Framework of Reference for Languages, ranging from A1
(no use of digital technologies in the classroom) to C2 (champion the use
of digital technologies in teaching environments, actively contribute

> to the stack of available digital tools and curate open educational resources).

Reviewer 3: again, not quite the right word. stack implies order and is not interconnected

Author: What about "to the palette of available digital tools" ? "palette" already conveys the concept of a range of available choices, in this case of "digital tools".

Reviewer: Better for me, but still palette lacks the interconnectivity and the different things that work together like compiler, IDE, formatter, containers, VCS, CI, the tons of imports (pandas, numpy,...), intelliSense,...

Author: @mschwarzmeier @mhagdorn Then what about
`- actively contribute to the ecosystem of digital tools and curate open educational resources`
`+ actively contribute to interoperable digital tools and related open educational resources`
?
In the above suggestion, _interoperable_ is the best relation one may wish from the tools I guess.

Reviewer: is better than stack or palette

> to the stack of available digital tools and curate open educational resources).

Reviewer 2: ```suggestion
to the palette of interoperable digital tools and curate open educational resources).
```
